buckets/local/archive.go

```diff
@@ -176,131 +175,52 @@ type ArchiveMessageType int
 const (
 	// ArchiveMessage accompanies an informational message.
 	ArchiveMessage ArchiveMessageType = iota
-	// ArchiveWarning accompanies a warning state.
-	ArchiveWarning
 	// ArchiveError accompanies an error state.
 	ArchiveError
-	// ArchiveSuccess accompanies a successful state.
-	ArchiveSuccess
 )
 
-// ArchiveStatus returns the current archive status.
-// When watch is true, the channel remains open, delivering all messages.
-func (b *Bucket) ArchiveStatus(ctx context.Context, watch bool) (<-chan ArchiveStatusMessage, error) {
+// Archives returns information about current and historical archives.
+func (b *Bucket) Archives(ctx context.Context) (*pb.ArchivesResponse, error) {
 	b.Lock()
 	defer b.Unlock()
 	ctx, err := b.context(ctx)
 	if err != nil {
 		return nil, err
 	}
 	key := b.Key()
-	rep, err := b.clients.Buckets.ArchiveStatus(ctx, key)
+	return b.clients.Buckets.Archives(ctx, key)
+}
+
+// ArchiveWatch delivers messages about the archive status.
+func (b *Bucket) ArchiveWatch(ctx context.Context, watch bool) (<-chan ArchiveStatusMessage, error) {
```

Comment:
I decided to simplify the old pattern of "get status optionally watch by providing a flag" because the types of data returned by status (now `Archives`) and `ArchiveWatch` are very different.